generated/model.py

```diff
@@ -1,6 +1,6 @@
 # generated by datamodel-codegen:
 #   filename:  public-api.yaml
-#   timestamp: 2023-11-09T05:00:29+00:00
+#   timestamp: 2024-01-04T19:49:36+00:00
```

Comment:
Initially I thought I would need to regenerate the SDK code, so I ran `make generate`, which is why the timestamp here is changed. Turns out this PR doesn't need to change the API spec.

Reply:
In this case we don't actually need to add anything to `public-api.yaml`. It is sufficient to just add the metadata field to the `DetectorCreationInput` object, which is what I'm doing.

Reply:
You can also see that the tests are passing without making any modifications to the API spec.

Reply:
I think we want to change the `api-spec.yaml`, though, for consistency. I see that it works without it, but we'd rather have things consistent between code and spec.